packages/flutter_tools/lib/src/http_host_availability_validator.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.9
+
+import 'base/io.dart';
+import 'base/platform.dart';
+import 'doctor.dart';
+import 'features.dart';
+
+// The environment variables used to override some URLs
+const String kPubHostedUrl = 'PUB_HOSTED_URL';
+const String kCloudUrl = 'FLUTTER_STORAGE_BASE_URL';
+
+/// Validation class that checks if all the given URLs are reachable
+class HttpHostAvailabilityValidator extends DoctorValidator {
+  HttpHostAvailabilityValidator({
+    Platform platform,
```

Comment:
These fields should be annotated as `@required`

Reply:
Fixed, thank you!